### ISSUE TITLE
Problems with windows explorer shell

### ISSUE THREAD

**Issue by conkerts** [ChrisTitusTech/winutil#306]
**Describe the bug**
This is really really difficult to describe. I never experienced something like that before.

So I have this extremely weird glitch with windoze explorer and the whole desktop UI, that various windows just start randomly flashing and hanging. Like if you move sometimes over the X and it starts changing color as if you hovering over it and then move your cursor away (but you're actually not moving your cursor at all, it just does it by itself).

Also feels like it is maybe trying to load some stuff from somewhere (registry/classes/dlls whatever), but it's missing, and that's why it ends up in this weird flashing loop and hang.

All my opened windows (esp. from brave browser) get randomly rescaled and moved outside the screen. I have to do shift+right click on taskbar and maximize, to get them back. Extremely annoying.

Hwinfo is also a tool that is severely affected and randomly freezes and the windows just disappear and reappear again.

**To Reproduce**
Use windoze explorer and after a while it starts happening.
Same with hwinfo and Telegram.
Other apps seem less affected

**Expected behavior**
Obviously no bugs/no flickering/hanging/window resizing

**Screenshots**
 At least that I could capture this one here:
https://cdn.discordapp.com/attachments/876538388801519676/1030924844570771486/video_2022-10-15_23-25-27.mp4

**Additional context**
I also don't want to accuse you/your tool of anything, but this started at a time when I tried it the first time, so maybe there is a correlation. 
(But also I did the September windoze updates at the end of September, so maybe those caused it, too. 
I mostly exclude drivers as a cause bc I haven't updated any drivers for a few months before September, tho in October I updated all drivers, just to see if the bug disappears. Surprise: It didn't)

I did click create a system restore point, but by the time I got the idea to try it, it seems it was already gone.

I also got SO FED UP with this bug, that I am just trying now cairoshell and the bug seems to be gone. (except when i try to use explorer windows to manage files)

**Comment by xfury94** [comment]
Also having this issue, did you ever found a solution without cairo?

**Comment by conkerts** [comment]
@xfury94 oh no, it is absolutely not a real solution with cairo. It just makes this piece of s**t borderline bearable.
Many windows still randomly change their size/get moved it seems, and sometimes I still see this weird flashing, so dwm (or whatever is drawing the title bar) is still affected.

I haven't delved deep into any windoze versions after xp/w7, so I still know some old tricks, but I have never encountered anything like this. 
Honestly as this windoze install is already over 1,5years old, it is probably better to just delete/reinstall it but I am not really motivated to do it and waste even more time with it.
I just hoped maybe someone has similar issues and has some idea/quick fix.

For now I think this is beyond repair (esp with no restore point) and I will probably send this dumb OS back to hell where it belongs and will use this a motivation to  switch to Linux finally.

But yeah, any ideas what to try out are welcome, I don't really care if I destroy this windoze install anymore.
That's why I am also trying a few things like these shells before I nuke this install.

**Comment by xfury94** [comment]
Already found the solution, edit winutil.ps1, remove everything related to timeouts and it's working fine.

There are some errors in the script like MenuShowDelay that is setting a DWord but on a stock instalation should be a string type.

**Comment by conkerts** [comment]
Hey, you are right !
At first I thought, well how can that help me, I don't have the default values anymore.
Though then again I just did some googling for the specific keys,
then some general search for hungapptimeout flickering issue
and then found these older forum posts:

https://www.sevenforums.com/general-discussion/239972-windows-7-window-flicker-not-screen-poor-mouse-response-aero-3.html
https://forums.tomshardware.com/threads/rapid-not-responding-flicker-random-tabbing-issues.1938122/post-15678153

To fix the screen flashing issue, follow steps below.
Run Registry Editor, and then delete the following registry keys


HKEY_USERS\.DEFAULT\Control Panel\Desktop\HungAppTimeOut
HKEY_USERS\.DEFAULT\Control Panel\Desktop\WaitToKillAppTimeOut
HKEY_CURRENT_USER\Control Panel\Desktop\HungAppTimeOut
HKEY_CURRENT_USER\Control Panel\Desktop\WaitToKillAppTimeOut
HKEY_CURRENT_USER\Control Panel\Desktop\WaitToKillServiceTimeOut

So yeah I just deleted them, and will check if that helps, but probably yes.
Also while looking them up, most of them are actually String and not Dword, so @ChrisTitusTech I'd also recommend more thorough checking of the datatypes for those registry hacks.
Kinda reminds me, why there was this advice to stay away from those in the first place ...

(And yes, a dword sounds more plausible to me to, but all the sites I checked actually said string, so I guess it's again some weird m$ bs that makes no sense ...)

**Comment by ChrisTitusTech** [comment]
I will make additional checks and remove these tweaks. I'll also go in and remove them if they were added. 
Can you verify that fixes your issue after removing the registry entries?

**Comment by xfury94** [comment]
Hi chris, for me it fixed the issue in two diferent machines, btw I was using some of your code in my personal script and found out a issue regarding svchost grouping, for some reason it presevents xboxgipsvc from starting and by proxy triggering gamebar with a controller.

I tried changing the value and even using SvcHostSplitDisable but no luck, the only fix was really reverting it to stock.

**Comment by ChrisTitusTech** [comment]
Great to know I'll patch this by remove the tweaks and also do a check to see if the tweaks exist and remove them. 

> Hi chris, for me it fixed the issue in two diferent machines, btw I was using some of your code in my personal script and found out a issue regarding svchost grouping, for some reason it presevents xboxgipsvc from starting and by proxy triggering gamebar with a controller.
> 
> I tried changing the value and even using SvcHostSplitDisable but no luck, the only fix was really reverting it to stock.
> 
> Edit: Get-Service -Name "XboxGipSvc" | Set-Service -StartupType Automatic also didn't work.

**Comment by conkerts** [comment]
> I will make additional checks and remove these tweaks. I'll also go in and remove them if they were added. Can you verify that fixes your issue after removing the registry entries?

I did only change those
HKEY_CURRENT_USER\Control Panel\Desktop\HungAppTimeOut
HKEY_CURRENT_USER\Control Panel\Desktop\WaitToKillAppTimeOut
HKEY_CURRENT_USER\Control Panel\Desktop\WaitToKillServiceTimeOut
(bc i didn't have anything under HKEY_USERS.DEFAULT )

3h ago I rebooted, disabled cairo ...
and no bugs till now, so yes, I think this has fixed it for me.

Big thanks to @xfury94 for the hint, otherwise I would have probably hit and destroyed my laptop sooner or later out of anger.
And I can postpone the windoze reinstall.

**Comment by xfury94** [comment]
@conkerts Good to know It worked fine for you too.

**Comment by conkerts** [comment]
I just saw, there were 2 other issues with the same flickering problem.
I just haven't found them, bc I forgot to remove the is:open while searching. They had the solution in there, too.
Maybe even that one line is enough, but who knows, maybe the other 2 lines might cause odd behaviour in some other circumstances.

Also that definitely backfired, that I did run the script in the beginning of Sept, and then haven't rebooted for around 22 days. So it was really hard to track back the problem.

**Comment by xfury94** [comment]
Similar thing happen to me, using this script for ages with no issues and all of a sudden flickering everywhere, after reading some insider changelogs I found that some people had issues with flickering so..... Maybe during that time between Microsoft fixing the issue and 22h2 release something changed with the timeouts.

**Comment by conkerts** [comment]
This flickering issue was super weird though.
I mean I had it in the beginning.
> Also feels like it is maybe trying to load some stuff from somewhere (registry/classes/dlls whatever), but it's missing, and that's why it ends up in this weird flashing loop and hang.

Idk how I came up with that, but I guess I still remember some stuff about odd things in windoze.
It certainly felt that way, that explorer trying to load something and just can't.

Maybe it's not even the value itself but the datatype that causes the problem/hickup. 
I mean I am too lazy to try that out now, and I honestly don't see much value in changing that one parameter (from what I read, it should be actually only used when shutting down?)
But anyway, all this tracking back and investigating probably took more of my time, than those speedups will ever recover :P 
On the other hand, I learned a few things again ...
